### PR TITLE
Added resources on data collection

### PR DESCRIPTION
**Pull request by jeulufis**
image

### CODE REVIEW

File: src/data/roadmaps/aws/content/100-introduction/103-aws/102-well-architected.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
...
 
-AWS Well-Architected Framework is a set of strategic guidelines provided by Amazon Web Services. It is designed to provide high-performing and resilient systems while maintaining cost efficiency. The framework divides the architectural best practices across five pillars which include operational excellence, security, reliability, performance efficiency, and cost optimization. With this framework, you can assess and improve your cloud-based architectures and applications by leveraging AWS technologies.
\ No newline at end of file
+AWS Well-Architected Framework is a set of strategic guidelines provided by Amazon Web Services. It is designed to provide high-performing and resilient systems while maintaining cost efficiency. The framework divides the architectural best practices across six pillars which include operational excellence, security, reliability, performance efficiency, cost optimization and the sustainability. With this framework, you can assess and improve your cloud-based architectures and applications by leveraging AWS technologies. 
+
+Visit the following resources to learn more:
+
+- [AWS Well-Architected](https://aws.amazon.com/architecture/well-architected)
```
Comment (dansholds) [inline src/data/roadmaps/aws/content/100-introduction/103-aws/102-well-architected.md:7]:
```suggestion
- [@article@AWS Well-Architected](https://aws.amazon.com/architecture/well-architected)
```

File: src/data/roadmaps/backend/content/120-real-time-data.md [hunk 2]
```diff
@@ -5,4 +5,7 @@ There are many ways to get real time data from the backend. Some of them are:
 - Websockets
 - Server Sent Events
 - Long Polling
-- Short Polling
\ No newline at end of file
+- Short Polling
+
+- [Introduction to HTTP Polling and Web Sockets](https://www.youtube.com/watch?v=OsgrJDMPl58)
+- [Introduction to Long Polling](https://www.pubnub.com/guides/long-polling/)
```
Comment (dansholds) [inline src/data/roadmaps/backend/content/120-real-time-data.md:11]:
```suggestion
- [@video@Introduction to HTTP Polling and Web Sockets](https://www.youtube.com/watch?v=OsgrJDMPl58)
- [@article@Introduction to Long Polling](https://www.pubnub.com/guides/long-polling/)
```

File: src/data/roadmaps/data-analyst/content/100-introduction/102-keyconcepts-for-data/100-collection.md [hunk 3]
```diff
@@ -1,3 +1,8 @@
...
+
+Visit the following resources to learn more:
+
+- [What Is Data Collection: Methods, Types, Tools](https://www.simplilearn.com/what-is-data-collection-article#what_is_data_collection)
+- [Data Collection Methods: Types & Examples](https://www.questionpro.com/blog/data-collection-methods/)
```
Comment (dansholds) [inline src/data/roadmaps/data-analyst/content/100-introduction/102-keyconcepts-for-data/100-collection.md:8]:
```suggestion
- [@article@What Is Data Collection: Methods, Types, Tools](https://www.simplilearn.com/what-is-data-collection-article#what_is_data_collection)
- [@article@Data Collection Methods: Types & Examples](https://www.questionpro.com/blog/data-collection-methods/)
```

File: src/data/roadmaps/react-native/content/104-core-components/100-text.md [hunk 4]
```diff
@@ -36,4 +36,8 @@ const styles = StyleSheet.create({
 export default SimpleText;
 ```
 
-In this example, we create a `Text` element with some text content, apply styling, limit it to two lines, and add an `onPress` event.
\ No newline at end of file
+In this example, we create a `Text` element with some text content, apply styling, limit it to two lines, and add an `onPress` event.
+
+Visit the following resources to learn more:
+
+- [Text - React Native](https://reactnative.dev/docs/text)
```
Comment (dansholds) [inline src/data/roadmaps/react-native/content/104-core-components/100-text.md:43]:
```suggestion
- [@official@Text - React Native](https://reactnative.dev/docs/text)
```

File: src/data/roadmaps/react-native/content/104-core-components/104-image-background.md [hunk 5]
```diff
@@ -35,4 +35,8 @@ const styles = StyleSheet.create({
 export default App;
 ```
 
-In the above example, `source` prop is used to add the image URL, `style` prop for adding some custom styling, and `resizeMode` to define how the image should stretch to fill the available space. The `Text` component inside the `ImageBackground` is then rendered on top of the image.
\ No newline at end of file
+In the above example, `source` prop is used to add the image URL, `style` prop for adding some custom styling, and `resizeMode` to define how the image should stretch to fill the available space. The `Text` component inside the `ImageBackground` is then rendered on top of the image.
+
+For more information and details, you can refer to the React Native docs:
+
+- [Image Background - React Native](https://reactnative.dev/docs/imagebackground)
```
Comment (dansholds) [inline src/data/roadmaps/react-native/content/104-core-components/104-image-background.md:42]:
```suggestion
Visit the following resources to learn more:

- [@official@Image Background - React Native](https://reactnative.dev/docs/imagebackground)
```

File: src/data/roadmaps/rust/content/105-traits-generics/100-trait-def.md [hunk 6]
```diff
@@ -1,3 +1,7 @@
...
\ No newline at end of file
+A `trait` definition in Rust is a way to define a set of behaviors necessary for a certain type. It is essentially an interface that types can implement. The `trait` def is created using the `trait` keyword followed by its name and the set of methods it includes enclosed in curly brackets. These methods are defined under the trait with their signature but without their implementation. Once a trait is defined, it can be implemented for any data type. Note that the type that this trait applies to is represented by the keyword `Self`. For example, `trait GetName { fn get_name(&self) -> String; }` defines a trait `GetName` with a method `get_name`. This trait can then be implemented for any type that needs the behaviour `get_name`.
+
+Visit the following resources to learn more:
+
+- [Rust by Example: Traits](https://doc.rust-lang.org/rust-by-example/trait.html)
```
Comment (dansholds) [inline src/data/roadmaps/rust/content/105-traits-generics/100-trait-def.md:7]:
```suggestion
- [@official@Rust by Example: Traits](https://doc.rust-lang.org/rust-by-example/trait.html)
```

File: src/data/roadmaps/rust/content/105-traits-generics/102-advanced-generics.md [hunk 7]
```diff
@@ -1,3 +1,8 @@
...
\ No newline at end of file
+Advanced generics in Rust offer a powerful set of tools for creating reusable and efficient code. The `where` syntax in generics can be used to specify trait and lifetime bounds, creating a more expressive declaration and avoiding numerous `T: Trait` inline annotations. Additionally, the `<T: ?Sized>` syntax allows for using dynamically sized types in generics. Advanced generics also allow for defining methods that apply to a subset of variations of generic types using an associated type. Furthermore, this allows for operator overloading through the use of associated types in traits, enhancing flexibility while retaining strong typing. Another advanced use of generics is 'higher kinded types', allowing a type constructor to take another type constructor as a parameter. This is the system Rust uses for handling `Option<T>`, `Result<T, E>`, and other similar types.
+
+Visit the following resources to learn more:
+
+- [Book: Generics](https://doc.rust-lang.org/book/ch10-01-syntax.html)
+- [Rust by Example: Generics](https://doc.rust-lang.org/rust-by-example/generics.html)
```
Comment (dansholds) [inline src/data/roadmaps/rust/content/105-traits-generics/102-advanced-generics.md:8]:
```suggestion
- [@official@Book: Generics](https://doc.rust-lang.org/book/ch10-01-syntax.html)
- [@official@Rust by Example: Generics](https://doc.rust-lang.org/rust-by-example/generics.html)
```

File: src/data/roadmaps/rust/content/105-traits-generics/index.md [hunk 8]
```diff
@@ -1,3 +1,7 @@
...
\ No newline at end of file
+Traits generics are a notable feature in Rust that pertain to defining shared behavior across data types. With them, you can specify placeholder types in trait definitions, letting you abstract over a range of possibilities. They're a means for defining shared behavior -- abstracting over functionality that types can have in common. You can apply traits to generics to constrain the types passed to the generic, hence permitting trait methods to be called on those types. You identify a generic parameter's type by the trait it implements, making it possible to use generic types in the trait methods. In essence, Rust achieves polymorphism through utilizing traits on generics.
+
+Visit the following resources to learn more:
+
+- [Book: Generic Types, Traits, and Lifetimes](https://doc.rust-lang.org/book/ch10-00-generics.html)
```
Comment (dansholds) [inline src/data/roadmaps/rust/content/105-traits-generics/index.md:7]:
```suggestion
- [@official@Generic Types, Traits, and Lifetimes](https://doc.rust-lang.org/book/ch10-00-generics.html)
```

File: src/data/roadmaps/rust/content/109-async-programming/100-tokio.md [hunk 9]
```diff
@@ -1,3 +1,8 @@
...
\ No newline at end of file
+Tokio, a Rust framework for developing applications, is used primarily for asynchronous programming that enables you to write asynchronous inputs/output, networking, and other features. Its primary function is to deliver high-performance, reliable, and easy-to-use asynchronous event-driven platform. It is built on the futures library and uses the async/await syntax of Rust for readability purposes. Tokio's 'runtime' provides I/O driver functions, scheduling and timers offering a foundation for asynchronous programming. It is designed to handle a high volume of network connections concurrently, making it ideal for building network applications.
+
+Visit the following resources to learn more:
+
+- [Official Website](https://tokio.rs/)
+- [Docs.rs: Tokio](https://docs.rs/tokio/latest/tokio/)
```
Comment (dansholds) [inline src/data/roadmaps/rust/content/109-async-programming/100-tokio.md:8]:
```suggestion
- [@official@Official Website](https://tokio.rs/)
- [@official@Docs.rs: Tokio](https://docs.rs/tokio/latest/tokio/)
```

File: src/data/roadmaps/rust/content/109-async-programming/101-async-std.md [hunk 10]
```diff
@@ -1,3 +1,7 @@
...
\ No newline at end of file
+`async-std` is a Rust library that provides an asynchronous version of the standard library. With the goal of being a drop-in replacement for Rust's standard library, it brings asynchronous programming directly into Rust's native system library, std. The most essential part inside `async-std` is an asynchronous runtime which includes IO and task scheduling. It lets you write asynchronous code that looks like synchronous code without having to worry about using future combinators or remembering to check if futures are ready. This significantly simplifies Rust's asynchronous programming model.
+
+Visit the following resources to learn more:
+
+- [Docs.rs: async-std](https://docs.rs/async-std/latest/async_std/)
```
Comment (dansholds) [inline src/data/roadmaps/rust/content/109-async-programming/101-async-std.md:7]:
```suggestion
- [@official@Docs.rs: async-std](https://docs.rs/async-std/latest/async_std/)
```

File: src/data/roadmaps/rust/content/109-async-programming/102-smol.md [hunk 11]
```diff
@@ -1,3 +1,7 @@
 # smol
 
-`smol` is a small, fast, and modern async runtime for Rust programming language. It is built on top of async-std and tokio. With very few lines of code, it allows users to perform tasks such as creating async functions, waiting on a future, creating a timer, among others. Although it is feature-rich, smol keeps its API minimal and clean, making it the go-to choice for many developers working with Rust. It supports async/.await natively and is highly efficient due to its superior scheduling capabilities.
\ No newline at end of file
+`smol` is a small, fast, and modern async runtime for Rust programming language. It is built on top of async-std and tokio. With very few lines of code, it allows users to perform tasks such as creating async functions, waiting on a future, creating a timer, among others. Although it is feature-rich, smol keeps its API minimal and clean, making it the go-to choice for many developers working with Rust. It supports async/.await natively and is highly efficient due to its superior scheduling capabilities.
+
+Visit the following resources to learn more:
+
+- [Docs.rs: smol](https://docs.rs/smol/latest/smol/)
```
Comment (dansholds) [inline src/data/roadmaps/rust/content/109-async-programming/102-smol.md:7]:
```suggestion
- [@official@Docs.rs: smol](https://docs.rs/smol/latest/smol/)
```

File: src/data/roadmaps/rust/content/119-testing/index.md [hunk 12]
```diff
@@ -1,3 +1,8 @@
...
\ No newline at end of file
+"Testing" in Rust is a crucial part of any programming project. This procedure involves creating specific scenarios to determine whether your code functions as expected. Rust has built-in support for this via the `cargo test` command which will run your test functions. These test functions are typically marked with the `#[test]` attribute to signify that they are not regular functions but testing adjuncts. Rust also provides a few macros such as `assert!`, `assert_eq!`, and `assert_ne!` for comparison checks and confirming expected behavior.
+
+Visit the following resources to learn more:
+
+- [Rust Book: Writing Automated Tests](https://doc.rust-lang.org/book/ch11-00-testing.html)
+- [YouTube](https://www.youtube.com/watch?v=8XaVlL3lObQ)
```
Comment (dansholds) [inline src/data/roadmaps/rust/content/119-testing/index.md:8]:
```suggestion
- [@official@Writing Automated Tests](https://doc.rust-lang.org/book/ch11-00-testing.html)
- [@video@Mocking Rust and Testing](https://www.youtube.com/watch?v=8XaVlL3lObQ)
```

File: src/data/roadmaps/rust/content/119-testing/101-mocking.md [hunk 13]
```diff
@@ -1,3 +1,9 @@
...
\ No newline at end of file
+In Rust, **mocking** is a process that lets you create fake functions, objects, or behaviors to test different conditions and scenarios in your code. Rust does not natively support mocking, however, there are external libraries present to help you perform mocking. Some of the popular libraries for mocking in rust are `mockito`, `mockall`, and `mockall_double`. These libraries give you the ability to create mock structures with the same API as your original code and allow you to set predefined responses from functions or validate function calls, thus helping you test your rust code thoroughly.
+
+Visit the following resources to learn more:
+
+- [Docs.rs: mockito](https://docs.rs/mockito/latest/mockito/)
+- [Docs.rs: mockall](https://docs.rs/mockall/latest/mockall/)
+- [Docs.rs: mockall_double](https://docs.rs/mockall_double/latest/mockall_double/)
```
Comment (dansholds) [inline src/data/roadmaps/rust/content/119-testing/101-mocking.md:9]:
```suggestion
- [@official@Docs.rs: mockito](https://docs.rs/mockito/latest/mockito/)
- [@official@Docs.rs: mockall](https://docs.rs/mockall/latest/mockall/)
- [@official@Docs.rs: mockall_double](https://docs.rs/mockall_double/latest/mockall_double/)
```

File: src/data/roadmaps/rust/content/119-testing/100-unit-integration.md [hunk 14]
```diff
@@ -1,3 +1,8 @@
...
\ No newline at end of file
+In Rust language, the concept of unit integration encompasses writing tests for individual units of your code, typically a function or method, as well as for multiple units that interact with each other. Unit tests are written in the same files as the code and are used to verify the functionality of a single unit of software in isolation. On the other hand, integration tests are stored in an entirely different directory and are meant to test how multiple components or modules of your application work together. They rely on combining units of code and testing the group, identifying issues that may not be visible when units are tested in isolation.
+
+Visit the following resources to learn more:
+
+- [Rust Book: How to Write Tests](https://doc.rust-lang.org/book/ch11-01-writing-tests.html)
+- [Rust by Example: Unit testing](https://doc.rust-lang.org/rust-by-example/testing/unit_testing.html)
```
Comment (dansholds) [inline src/data/roadmaps/rust/content/119-testing/100-unit-integration.md:8]:
```suggestion
- [@official@Rust Book: How to Write Tests](https://doc.rust-lang.org/book/ch11-01-writing-tests.html)
- [@official@Rust by Example: Unit testing](https://doc.rust-lang.org/rust-by-example/testing/unit_testing.html)
```

File: src/data/roadmaps/rust/content/110-networking/102-quinn.md [hunk 15]
```diff
@@ -1,3 +1,7 @@
...
\ No newline at end of file
+`Quinn` is a Rust networking library providing high-level access to the QUIC protocol. It is built on top of the `tokio` runtime and implements the QUIC transport protocol as specified by the IETF. It provides an async, futures-based API for working with QUIC connections and streams. It is designed to offer high performance with robustness and flexibility, supporting both client and server roles. The QUIC protocol itself is a multiplexed and secure transport protocol positioned as a modern alternative to TCP, offering better performance for many networking applications.
+
+Visit the following resources to learn more:
+
+- [Docs.rs: quinn](https://docs.rs/quinn/latest/quinn/)
```
Comment (dansholds) [inline src/data/roadmaps/rust/content/110-networking/102-quinn.md:7]:
```suggestion
- [@official@Docs.rs: quinn](https://docs.rs/quinn/latest/quinn/)
```

File: src/data/roadmaps/rust/content/110-networking/101-hyper.md [hunk 16]
```diff
@@ -1,3 +1,8 @@
...
\ No newline at end of file
+"Hyper" is a fast, safe and concurrent HTTP client and server written in Rust. It uses a non-blocking I/O model for efficient execution, leverages the Tokio platform for event-driven, asynchronous I/O, and includes support for HTTP/2. It has a modular design, allowing you to choose the features you need. Hyper also features a server-side framework for building your own HTTP applications. For security and speed, it comes with native support for HTTP/1 and HTTP/2, and automatically negotiates these protocols. Most importantly, it is designed from the ground up to take advantage of Rust's memory safety and concurrency features.
+
+Visit the following resources to learn more:
+
+- [Official Website](https://hyper.rs/)
+- [Docs.rs: Hyper](https://docs.rs/hyper/latest/hyper/)
```
Comment (dansholds) [inline src/data/roadmaps/rust/content/110-networking/101-hyper.md:8]:
```suggestion
- [@official@Official Website](https://hyper.rs/)
- [@official@Docs.rs: Hyper](https://docs.rs/hyper/latest/hyper/)
```

File: src/data/roadmaps/rust/content/110-networking/100-reqwest.md [hunk 17]
```diff
@@ -1,3 +1,7 @@
...
\ No newline at end of file
+`Reqwest` is a Rust library which is designed to make HTTP requests easy and effortless. It offers a mix of convenience methods for both synchronous and asynchronous requests,GET,POST and other HTTP methods. It also supports JSON content and is built on `hyper` for HTTP and `tokio` for asynchronous I/O. This means you can take advantage of the powerful asynchronous features of Rust such as Futures and async/await. Furthermore, `Reqwest` takes care of many tedious aspects of HTTP for you, such as handling cookies and encoding/decoding different formats.
+
+Visit the following resources to learn more:
+
+- [Docs.rs: Reqwest](https://docs.rs/reqwest/latest/reqwest/)
```
Comment (dansholds) [inline src/data/roadmaps/rust/content/110-networking/100-reqwest.md:7]:
```suggestion
- [@official@Docs.rs: Reqwest](https://docs.rs/reqwest/latest/reqwest/)
```

**Comment by dansholds** [comment]
@jeulufis Please fix the merge conflicts before we merge.

Also read our contributions guidelines as you did not apply any content tags.

**Comment by jeulufis** [comment]
How can I fix fusion conflicts? , sorry for the question, I am new to open source.

**Comment by dansholds** [comment]
@jeulufis I would suggest you take note of the changes you made, pull the latest from master, add the changes you wanted to and then commit them all in one clean commit ☺️

**Pull request closed without merging**